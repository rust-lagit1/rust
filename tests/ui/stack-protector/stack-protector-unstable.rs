//@ revisions: all strong strong-ok basic basic-ok
//@ compile-flags: --target x86_64-unknown-linux-gnu
//@ needs-llvm-components: x86
//@ [all] check-pass
//@ [all] compile-flags: -C stack-protector=all
//@ [strong] check-fail
//@ [strong] compile-flags: -C stack-protector=strong
//@ [strong-ok] check-pass
//@ [strong-ok] compile-flags: -C stack-protector=strong -Z unstable-options
//@ [basic] check-fail
//@ [basic] compile-flags: -C stack-protector=basic
//@ [basic-ok] check-pass
//@ [basic-ok] compile-flags: -C stack-protector=basic -Z unstable-options

#![crate_type = "lib"]
#![feature(no_core, lang_items)]
#![no_std]
#![no_core]

#[lang = "sized"]
trait Sized {}
#[lang = "copy"]
trait Copy {}

pub fn main(){}
