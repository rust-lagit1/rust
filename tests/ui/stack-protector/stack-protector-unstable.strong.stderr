error: `-C stack-protector=basic` and `-C stack-protector=strong` require `-Z unstable-options`

